src/libcollections/str.rs

```diff
@@ -552,13 +552,13 @@ impl str {
 
     /// Returns a slice of the string from the character range [`begin`..`end`).
     ///
-    /// That is, start at the `begin`-th code point of the string and continue
-    /// to the `end`-th code point. This does not detect or handle edge cases
-    /// such as leaving a combining character as the first code point of the
+    /// That is, start at the `begin`-th Unicode scalar value of the string and continue
+    /// to the `end`-th Unicode scalar value. This does not detect or handle edge cases
+    /// such as leaving a combining character as the first `char` of the
     /// string.
     ///
     /// Due to the design of UTF-8, this operation is `O(end)`. Use slicing
-    /// syntax if you want to use byte indices rather than codepoint indices.
+    /// syntax if you want to use byte indices rather than `char` indices.
```

Comment:
Oh this is confusing -- we have an iterator called `char_indices()` which returns byte offsets from the start of the string paired with `char`.